modules.cmake

```diff
@@ -69,16 +69,21 @@ endfunction()
 # Checks that the compiler supports C++20 modules.
 #
 # Usage:
-#   modules_supported(<variable_name>)
+#   modules_supported(<variable_name> [STANDARD standard_ver])
```

Comment:
I suggest checking `CMAKE_CXX_STANDARD` in `modules_supported` directly instead of asking the caller to pass it. If there is a use case for not checking it we can add an opt out later but checking seems like the right default.

Reply:
Reworked in a following way

- no args
  - use `${CMAKE_CXX_STANDARD}`
  - in case `NOT DEFINED ${CMAKE_CXX_STANDARD}` use the latest available
 - `STANDARD` argument provided
  - check for the specified standard

Reply:
Seems reasonable.